pkg/registry/registry.go

```diff
@@ -206,22 +208,46 @@ func RetagImage(repo, tag, srcRegistry, dstRegistry string) error {
 }
 
 func (r *Registry) RepositoryTagExists(repo, tag string) (bool, error) {
-	endpoint := fmt.Sprintf("%s/v1/repositories/%s/tags/%s", r.address, repo, tag)
-	req, err := http.NewRequest("GET", endpoint, nil)
-	if err != nil {
-		return false, microerror.Mask(err)
-	}
+	var imageExists bool
 
-	resp, err := r.httpClient.Do(req)
-	if err != nil {
-		return false, microerror.Mask(err)
-	}
+	image := fmt.Sprintf("%s/%s:%s", r.name, repo, tag)
 
-	if resp.StatusCode == 200 {
-		return true, nil
-	}
+	args := []string{"pull", image}
+
+	cmd := exec.Command(
+		dockerBinaryName,
+		args...,
+	)
+
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+
+	go func() {
+		_ = cmd.Run()
+	}()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		for outb.String() == "" && errb.String() == "" {
+			time.Sleep(time.Millisecond * 10)
+		}
```

Comment:
It would be good to timeout. Maybe use `backoff` lib.

Reply:
with separate pr

Reply:
original docker pull command has a timeout so there is no chance it will not exist with something written into stdout or stderr